RecoLocalTracker/SiPixelClusterizer/plugins/SiPixelPhase2DigiToClusterCUDA.cc

```diff
@@ -0,0 +1,156 @@
+
+class SiPixelPhase2DigiToClusterCUDA : public edm::stream::EDProducer<edm::ExternalWork> {
+public:
+  explicit SiPixelPhase2DigiToClusterCUDA(const edm::ParameterSet& iConfig);
+  ~SiPixelPhase2DigiToClusterCUDA() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void acquire(const edm::Event& iEvent,
+               const edm::EventSetup& iSetup,
+               edm::WaitingTaskWithArenaHolder waitingTaskHolder) override;
+  void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+
+  const edm::EDGetTokenT<edm::DetSetVector<PixelDigi>> pixelDigiToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+
+  edm::EDPutTokenT<cms::cuda::Product<SiPixelDigisCUDA>> digiPutToken_;
+  edm::EDPutTokenT<cms::cuda::Product<SiPixelDigiErrorsCUDA>> digiErrorPutToken_;
+  edm::EDPutTokenT<cms::cuda::Product<SiPixelClustersCUDA>> clusterPutToken_;
+
+  cms::cuda::ContextState ctxState_;
+
+  pixelgpudetails::SiPixelRawToClusterGPUKernel gpuAlgo_;
+  std::unique_ptr<pixelgpudetails::SiPixelRawToClusterGPUKernel::WordFedAppender> wordFedAppender_;
+
+  const bool includeErrors_;
+  const SiPixelClusterThresholds clusterThresholds_;
+};
+
+SiPixelPhase2DigiToClusterCUDA::SiPixelPhase2DigiToClusterCUDA(const edm::ParameterSet& iConfig)
+    : pixelDigiToken_(consumes<edm::DetSetVector<PixelDigi>>(iConfig.getParameter<edm::InputTag>("InputDigis"))),
+      digiPutToken_(produces<cms::cuda::Product<SiPixelDigisCUDA>>()),
+      clusterPutToken_(produces<cms::cuda::Product<SiPixelClustersCUDA>>()),
+      includeErrors_(iConfig.getParameter<bool>("IncludeErrors")),
+      clusterThresholds_{iConfig.getParameter<int32_t>("clusterThreshold_layer1"),
+                         iConfig.getParameter<int32_t>("clusterThreshold_otherLayers")} {
+  if (includeErrors_) {
+    digiErrorPutToken_ = produces<cms::cuda::Product<SiPixelDigiErrorsCUDA>>();
+  }
+}
+
+void SiPixelPhase2DigiToClusterCUDA::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<bool>("IncludeErrors", true);
+  desc.add<int32_t>("clusterThreshold_layer1", kSiPixelClusterThresholdsDefaultPhase2.layer1);
+  desc.add<int32_t>("clusterThreshold_otherLayers", kSiPixelClusterThresholdsDefaultPhase2.otherLayers);
+  desc.add<edm::InputTag>("InputDigis", edm::InputTag("simSiPixelDigis:Pixel"));
+  descriptions.addWithDefaultLabel(desc);
+}
+
+void SiPixelPhase2DigiToClusterCUDA::acquire(const edm::Event& iEvent,
+                                             const edm::EventSetup& iSetup,
+                                             edm::WaitingTaskWithArenaHolder waitingTaskHolder) {
+  cms::cuda::ScopedContextAcquire ctx{iEvent.streamID(), std::move(waitingTaskHolder), ctxState_};
+
+  auto const& input = iEvent.get(pixelDigiToken_);
+
+  const TrackerGeometry* geom_ = &iSetup.getData(geomToken_);
```

Comment:
The `esConsumes()` is missing for `geomToken_` and that is causing workflow 39434.502 step 3 to fail on a machine with a GPU (see https://github.com/cms-sw/cmssw/issues/36604).